src/redux/data.js

```diff
@@ -128,6 +143,141 @@ export const dataLoadState = () => async (dispatch, getState) => {
   } catch (error) {
     dispatch({ type: DATA_LOAD_TRANSACTIONS_FAILURE });
   }
+  genericAssetsHandle = setTimeout(() => {
+    dispatch(genericAssetsFallback());
+  }, GENERIC_ASSETS_FALLBACK_TIMEOUT);
+};
+
+export const fetchAssetPrices = async (coingeckoIds, nativeCurrency) => {
+  try {
+    const url = `https://api.coingecko.com/api/v3/simple/price?ids=${coingeckoIds
+      .filter(val => !!val)
+      .sort()
+      .join(
+        ','
+      )}&vs_currencies=${nativeCurrency}&include_24hr_change=true&include_last_updated_at=true`;
+    const priceRequest = await fetch(url);
+    return priceRequest.json();
+  } catch (e) {
+    logger.log(`Error trying to fetch ${coingeckoIds} prices`, e);
+  }
+};
+
+const genericAssetsFallback = () => async (dispatch, getState) => {
+  logger.log('ZERION IS DOWN! ENABLING GENERIC ASSETS FALLBACK');
+  const { nativeCurrency } = getState().settings;
+  const formattedNativeCurrency = toLower(nativeCurrency);
+  let ids;
+  try {
+    const request = await fetch(COINGECKO_IDS_ENDPOINT);
+    ids = await request.json();
+  } catch (e) {
+    ids = coingeckoIdsFallback;
+  }
+
+  const allAssets = [
+    {
+      asset_code: 'eth',
+      coingecko_id: 'ethereum',
+      decimals: 18,
+      name: 'Ethereum',
+      symbol: 'ETH',
+    },
+    {
+      asset_code: DPI_ADDRESS,
+      coingecko_id: 'defipulse-index',
+      decimals: 18,
+      name: 'DefiPulse Index',
+      symbol: 'DPI',
+    },
+  ];
+
+  keys(TokensListenedCache).forEach(address => {
+    const coingeckoAsset = ids.find(
+      ({ platforms: { ethereum: tokenAddress } }) =>
+        toLower(tokenAddress) === toLower(address)
+    );
+
+    if (coingeckoAsset) {
+      allAssets.push({
+        asset_code: address,
```

Comment:
does address also need to be `toLower(address)` here (or are we adding addresses to the TokensListenedCache already with lowered address, in which case we don't need the `toLower(address)` in line 198)

Reply:
We're. Removing the check on line 198